src/app/demo/demo.component.spec.ts

```diff
@@ -239,3 +238,104 @@ function randomRange(min: number, max: number) {
+    SBOIEcgWSBVVyBOIj5mZW5neXVuPC93PiA8dyBpZD0idDBiMGQwcDBzMHc5IiBlZmZlY3RpdmUtZzJwLWxhbmc9InVuZCIgQVJQQ
+    UJFVD0iTSBFWSBOIEcgRCBJWSI+bWVuZ2RpPC93PiA8dyBpZD0idDBiMGQwcDBzMHcxMCIgQVJQQUJFVD0iRUggUiBJSCBLIj5lc
+    mljPC93Pjwvcz48cCBjbGFzcz0idHJhbnNsYXRpb24iPui/meaYr+S4gOS4qua1i+ivlea2jOWQkeiSmeiSgm48L3A+CiAgICAgI
+    CAgICAgICAgICA8L3A+CiAgICAgICAgICAgIDwvZGl2PgogICAgICAgIDwvYm9keT4KICAgIDwvdGV4dD4KPC9URUk+`;
+    let translation = service.b64_to_utf8(testxml);
+    expect(service.b64_to_utf8(service.utf8_to_b64(testxml))).toEqual(testxml);
+  });
+
+  // the situation when the sentence ocupies more than one line
+  it("should b63service works with a medium length translation", () => {
+    let testxml = `PFRFST4KICAgIDx0ZXh0IHhtbDpsYW5nPSJlbmciIGZhbGxiYWNrLWxhbmdzPSJ1bmQiIGlkPSJ0MCI+CiAgICAgICAgPGJvZHkgaWQ9InQwY
+    jAiPgogICAgICAgICAgICA8ZGl2IHR5cGU9InBhZ2UiIGlkPSJ0MGIwZDAiPgogICAgICAgICAgICAgICAgPHAgaWQ9InQwYjBkMHAwIj4KICAgICAgICAgICAgI
+    CAgICAgICA8cyBpZD0idDBiMGQwcDBzMCI+PHcgaWQ9InQwYjBkMHAwczB3MCIgZWZmZWN0aXZlLWcycC1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XIE4gR
+    yBBQSI+Y2hvbmdhPC93PiA8dyBpZD0idDBiMGQwcDBzMHcxIiBlZmZlY3RpdmUtZzJwLWxhbmc9InVuZCIgQVJQQUJFVD0iQ0ggSEggT1cgTiBHIEFBIj5jaG9uZ
+    2E8L3c+IDx3IGlkPSJ0MGIwZDBwMHMwdzIiIGVmZmVjdGl2ZS1nMnAtbGFuZz0idW5kIiBBUlBBQkVUPSJDSCBISCBPVyBOIEcgQUEiPmNob25nYTwvdz4gPHcga
+    WQ9InQwYjBkMHAwczB3MyIgZWZmZWN0aXZlLWcycC1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XIE4gRyBBQSI+Y2hvbmdhPC93PiA8dyBpZD0idDBiMGQwc
+    DBzMHc0IiBlZmZlY3RpdmUtZzJwLWxhbmc9InVuZCIgQVJQQUJFVD0iQ0ggSEggT1cgTiBHIEFBIj5jaG9uZ2E8L3c+IDx3IGlkPSJ0MGIwZDBwMHMwdzUiIGVmZ
+    mVjdGl2ZS1nMnAtbGFuZz0idW5kIiBBUlBBQkVUPSJDSCBISCBPVyBOIEcgQUEiPmNob25nYTwvdz4gPHcgaWQ9InQwYjBkMHAwczB3NiIgZWZmZWN0aXZlLWcyc
+    C1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XIE4gRyBBQSI+Y2hvbmdhPC93PiA8dyBpZD0idDBiMGQwcDBzMHc3IiBlZmZlY3RpdmUtZzJwLWxhbmc9InVuZ
+    CIgQVJQQUJFVD0iQ0ggSEggT1cgTiBHIEFBIj5jaG9uZ2E8L3c+IDx3IGlkPSJ0MGIwZDBwMHMwdzgiIGVmZmVjdGl2ZS1nMnAtbGFuZz0idW5kIiBBUlBBQkVUP
+    SJDSCBISCBPVyBOIEcgQUEiPmNob25nYTwvdz4gPHcgaWQ9InQwYjBkMHAwczB3OSIgZWZmZWN0aXZlLWcycC1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XI
+    E4gRyBBQSI+Y2hvbmdhPC93PiA8dyBpZD0idDBiMGQwcDBzMHcxMCIgZWZmZWN0aXZlLWcycC1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XIE4gRyBBQSI+Y
+    2hvbmdhPC93PiA8dyBpZD0idDBiMGQwcDBzMHcxMSIgZWZmZWN0aXZlLWcycC1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XIE4gRyBBQSI+Y2hvbmdhPC93P
+    jwvcz48cCBjbGFzcz0idHJhbnNsYXRpb24iPmNob25nYSBjaG9uZ2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ
+    2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ2EgY2hvbmdhIGNob
+    25nYTwvcD4KICAgICAgICAgICAgICAgIDwvcD4KICAgICAgICAgICAgPC9kaXY+CiAgICAgICAgPC9ib2R5PgogICAgPC90ZXh0Pgo8L1RFST4=`;
+    let translation = service.b64_to_utf8(testxml);
+    expect(service.b64_to_utf8(service.utf8_to_b64(testxml))).toEqual(testxml);
+  });
+});
```

Comment:
After fixed, all items are great. Just one remainder, the first test is similar with the other component  test, But that fine. well done.

Reply:
It is all done for milestone3. It is great.